### PR TITLE
Ensure the correct culture is set in API controllers according to domain

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have [created an issue](https://github.com/umbraco/Umbraco-CMS/issues) for the proposed changes in this PR, the link is: https://github.com/umbraco/Umbraco-CMS/issues/3356
- [x] I have added steps to test this contribution in the description below

### Description

#3356 outlines the steps to reproduce.

An easy way to test this PR is to dump this API controller in App_Code:

```cshtml
using System;
using System.Threading;
using Umbraco.Web.WebApi;

namespace MySite
{
    public class TestCultureController : UmbracoApiController
    {
        public string GetCulture() {
          return Thread.CurrentThread.CurrentUICulture.Name;
        }
    }
}
```

Then call `/umbraco/api/testculture/getculture` to have it output the current thread culture. Verify that it changes according to the assigned domain culture in Umbraco.

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Hey @kjac - I've discussed this with @zpqrtbnk and there's a few things we can't overcome

- The code is a bit hacky (it works but.. feels wrong)
- Unit tests start breaking when this is applied 
- This will be a breaking change since people might be relying on the existing behavior
- This solution is only going to work when using domains... and the page you render may be in another language due to ... many things (you can override the language down the tree for example)
- Also doubt it's going to support domains such as "site.com/en"

In conclusion, this might solve some cases but we don't think it's the correct solution. The recommendation is to run the proposed code from your own custom code instead.
